packages/nimble-components/src/table/components/group-row/index.ts

```diff
@@ -0,0 +1,88 @@
+import { attr, html, observable, ViewTemplate } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import type { TableColumn } from '../../../table-column/base';
+import { styles } from './styles';
+import { template } from './template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-group-row': TableGroupRow;
+    }
+}
+
+/**
+ * A styled cell that is used within the nimble-table-row.
+ * @internal
+ */
+export class TableGroupRow extends FoundationElement {
+    @observable
+    public groupRowValue?: unknown;
+
+    @observable
+    public nestingLevel = 0;
+
+    @observable
+    public leafItemCount?: number;
+
+    @observable
+    public groupColumn?: TableColumn;
+
+    @attr({ mode: 'boolean' })
+    public expanded = false;
+
+    /**
+     * @internal
+     */
+    @observable
+    public groupHeaderViewTemplate?: ViewTemplate<TableGroupRow>;
+
+    /**
+     * @internal
+     */
+    public readonly expandIcon!: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public animationClass = '';
+
+    public onGroupExpandToggle(): void {
+        this.$emit('group-expand-toggle');
+        this.animationClass = 'animating';
+        this.expandIcon.addEventListener(
+            'transitionend',
+            this.removeAnimatingClass
+        );
+    }
+
+    public groupColumnChanged(): void {
+        if (this.groupColumn?.internalGroupHeaderViewTag) {
+            this.groupHeaderViewTemplate = html<TableGroupRow>`
```

Comment:
How often is this running? I imagine it is generally being set to the same value over and over again and not running often as you scroll? That relies on dirty checking needing to run over and over on the same column instance.

I mean this only ever needs to be calculated once per column when the associated TableColumn internalGroupHeaderViewTag changes.

Could we instead do something like:

```ts
class TableColumn {
   /* @internal */ @observable internalGroupHeaderViewTag;
   /* @internal */ @observable currentGroupHeaderViewTemplate;

   private internalGroupHeaderViewTagChanged() {
        this.currentGroupHeaderViewTemplate = createGroupHeaderViewTemplate(this.internalGroupHeaderViewTag);
   }
}

// Some other file:
export createGroupHeaderViewTemplate = (groupHeaderViewTag: string) => html<TableGroupRow>`
      <${groupHeaderViewTag}
          :groupHeaderValue="${x => x.groupRowValue}"
          :columnConfig="${x => x.groupColumn?.columnConfig}"
          class="group-header-value"
          >
      </${groupHeaderViewTag}>
  `;
```

The above guarantees we are re-running the template parser logic only when the tag name itself changes. (chose the name prefix `current` to emphasize that is a table managed value and not an api the plugin author should mess with).